source/Patches/Roles/Lover.cs

```diff
@@ -64,27 +64,24 @@ protected override string NameText(PlayerVoteArea player = null)
 
             return Player.name + "\n" + "Lover";
         }
-
+        
         public static void Gen(List<PlayerControl> crewmates, List<PlayerControl> impostors)
         {
-            //System.Console.WriteLine("LOVER2");
-            if (crewmates.Count <= 0) return;
-            if (crewmates.Count <= 1 && impostors.Count < 1) return;
-
-            //System.Console.WriteLine("LOVER3");
-            var b = Random.RandomRangeInt(0, 3);
-
-            if ((b == 0) & (impostors.Count < 1)) b = 1;
+            var lovingImpostorEnabled = Random.RandomRangeInt(1, 101) <= CustomGameOptions.LovingImpostorOn;
+            
+            var canMakeCrewCrewLovers = crewmates.Count >= 2;
+            var canMakeCrewImpostorLovers = crewmates.Count >= 1 && impostors.Count >= 2 && lovingImpostorEnabled;
```

Comment:
```suggestion
            var canMakeCrewImpostorLovers = crewmates.Count >= 1 && impostors.Count >= 1 && lovingImpostorEnabled;
```
there would only need to be one impostor free